sdk/identity/azure-identity/azure/identity/_credentials/azure_cli.py

```diff
@@ -68,14 +69,18 @@ def get_token(self, *scopes, **kwargs):  # pylint:disable=no-self-use,unused-arg
 def parse_token(output):
     """Parse output of 'az account get-access-token' to an AccessToken.
 
-    In particular, convert the CLI's "expiresOn" value, the string representation of a naive datetime, to epoch seconds.
+    In particular, convert the "expiresOn" value to epoch seconds. This value is a naive local datetime as returned by
+    datetime.fromtimestamp.
     """
     try:
         token = json.loads(output)
-        parsed_expires_on = datetime.strptime(token["expiresOn"], "%Y-%m-%d %H:%M:%S.%f")
-
-        # calculate seconds since the epoch; parsed_expires_on is naive
-        expires_on = (parsed_expires_on - datetime.fromtimestamp(0)).total_seconds()
+        dt = datetime.strptime(token["expiresOn"], "%Y-%m-%d %H:%M:%S.%f")
+        if hasattr(dt, "timestamp"):
+            # Python >= 3.3
+            expires_on = dt.timestamp()
```

Comment:
I don't really understand the original issue.

I tested on a UTC+8 machine and both forms return the same result. `datetime.fromtimestamp(0)` also returns a naive datetime of local time zone.

```py
from datetime import datetime

dt = datetime.strptime("2020-10-12 11:24:00.0000", "%Y-%m-%d %H:%M:%S.%f")
print((dt - datetime.fromtimestamp(0)).total_seconds())
print(dt.timestamp())
print(datetime.fromtimestamp(0))
```

Output:
```
1602473040.0
1602473040.0
1970-01-01 08:00:00
```

Reply:
A naive calculation doesn't consider daylight saving time. When daylight saving time is in effect, as it is today in Seattle, you see this:
```py
>>> dt = datetime.strptime("2020-10-12 11:24:00.0000", "%Y-%m-%d %H:%M:%S.%f")
>>> (dt - datetime.fromtimestamp(0)).total_seconds() - dt.timestamp()
3600.0
```

Reply:
Ah. I see. Interesting.

Reply:
Perhaps CLI should have stuck with UTC from the very beginning.

Reply:
It's too late to change "expiresOn", but a new field like "expiresOnUTC" could solve the problem.

Reply:
To add a little bit more context, `datetime.strptime("2020-10-12 11:24:00.0000", "%Y-%m-%d %H:%M:%S.%f")` returns a [naive](https://docs.python.org/3/library/datetime.html#aware-and-naive-objects) `datetime`, so `dt - datetime.fromtimestamp(0)` doesn't take daylight saving time into consideration.

Meanwhile, according to https://docs.python.org/3/library/datetime.html#datetime.datetime.timestamp,

> Naive [datetime](https://docs.python.org/3/library/datetime.html#datetime.datetime) instances are assumed to represent local time and this method relies on the platform C mktime() function to perform the conversion.

so `datetime.timestamp()` will take daylight saving time into consideration, thus being `3600` seconds smaller than the result of `dt - datetime.fromtimestamp(0)`.